imports/server/guesses.ts

```diff
@@ -86,6 +97,78 @@ Meteor.methods({
     });
   },
 
+  addCorrectGuessForPuzzle(puzzleId: unknown, answer: unknown) {
+    check(this.userId, String);
+    check(puzzleId, String);
+    check(answer, String);
+
+    const puzzle = Puzzles.findOne(puzzleId);
+
+    if (!puzzle) {
+      throw new Meteor.Error(404, 'No such puzzle');
+    }
+
+    const hunt = Hunts.findOne(puzzle.hunt);
+
+    if (!hunt) {
+      throw new Meteor.Error(404, 'No such hunt');
+    }
+
+    if (hunt.hasGuessQueue) {
+      throw new Meteor.Error(404, 'Hunt does not allow you to enter answers directly');
+    }
+
+    Ansible.log('New correct guess', {
+      hunt: puzzle.hunt,
+      puzzle: puzzleId,
+      user: this.userId,
+      guess: answer,
+    });
+    const answerId = Guesses.insert({
+      hunt: puzzle.hunt,
+      puzzle: puzzleId,
+      guess: answer,
+      state: 'correct',
+    });
+
+    const savedAnswer = Guesses.findOne(answerId);
+    if (!savedAnswer) {
+      throw new Meteor.Error(404, 'No such correct guess');
+    }
+    addChatMessage(savedAnswer, 'correct');
+    Puzzles.update({
+      _id: savedAnswer.puzzle,
+    }, {
+      $addToSet: {
+        answers: savedAnswer.guess,
+      },
+    });
+    GlobalHooks.runPuzzleSolvedHooks(savedAnswer.puzzle);
+  },
+
+  removeAnswerFromPuzzle(puzzleId: unknown, answer: unknown) {
```

Comment:
This feels like it'd be better taking a `guessId` than a puzzleId + answer, since the former is reliably unique and the latter is not.